dbms/src/Flash/ResourceControl/TokenBucket.cpp

```diff
@@ -85,7 +85,13 @@ double TokenBucket::getAvgSpeedPerSec()
 
 void TokenBucket::compact(const TokenBucket::TimePoint & timepoint)
 {
-    tokens += getDynamicTokens(timepoint);
+    auto dynamic_tokens = getDynamicTokens(timepoint);
+    RUNTIME_CHECK(dynamic_tokens >= 0.0);
+    // To avoid getDynamicTokens() too frequently.
```

Comment:
This comment makes me a little confused, why does return 0 here reduce the call frequency?

Reply:
If `TokenBucket::compact()` is called too frequent, like every 500ms, `dynamic_tokens` will always be zero, because less than 1 sec.

Reply:
I can update comments